src/flask_restalchemy/resources/resources.py

```diff
@@ -200,8 +207,9 @@ def get(self, relation_id, id=None):
                     'Use flask-sqlalchemy relationship with lazy="dynamic".')
                 collection = [self._serializer.dump(item) for item in relation_list_or_query]
             else:
-                collection = query_from_request(self._resource_model, self._serializer, request,
+                query = query_from_request(self._resource_model, self._serializer, request,
```

Comment:
query_callback should be called here too, and probably in all the places where `query_from_request` was being called